libs/runtime/unit/platform_unit_test.cpp

```diff
@@ -146,6 +146,59 @@ struct ebpf_hash_table_destroyer_t
 
 using ebpf_hash_table_ptr = std::unique_ptr<ebpf_hash_table_t, ebpf_hash_table_destroyer_t>;
 
+/**
+ * @brief A RAII class to enter and exit epoch.
+ */
+typedef class _ebpf_epoch_scope
+{
+  public:
+    /**
+     * @brief Construct a new ebpf epoch scope object and enter epoch.
+     */
+    _ebpf_epoch_scope() : in_epoch(false) { enter(); }
+
+    /**
+     * @brief Leave epoch if entered.
+     */
+    ~_ebpf_epoch_scope()
+    {
+        if (in_epoch) {
+            exit();
+        }
+    }
+
+    /**
+     * @brief Enter epoch.
+     */
+    void
+    enter()
+    {
+        if (in_epoch) {
```

Comment:
Just curious if this check (and subsequent setting to true) ought to be made atomic, i.e. some form of compareExchange?  was just wondering how this would behave in a multi-threaded scenario?

Reply:
Nope, this should only be called from a single thread. Epoch is very much thread specific, so calling enter/exit on another thread is undefined behavior / a bug.